src/main/java/net/minestom/server/event/AbstractPlayerChatEvent.java

```diff
@@ -0,0 +1,41 @@
+package net.minestom.server.event;
+
+import net.kyori.adventure.text.Component;
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.trait.PlayerEvent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public interface AbstractPlayerChatEvent extends PlayerEvent {
```

Comment:
Renaming to IPlayerChatEvent would probably be better for consistency

Reply:
I personally prefer `AbstractPlayerChatEvent`, but I am not sure about inheritance in events. Seems like a huge mess and there is likely a better solution to handle Pre/Post listeners

Reply:
So what should I do then?

Reply:
Something we discussed in discord was the ability to register an event as a "post" listener. Basically the event is fired and then the result is returned to the caller, at the same time a new thread is spun up to call the event for anyone listening to it in a "post" way. This way, we would add support for async monitoring for stuff like chat, movement, etc - without having to block the actual action itself.

Reply:
Which would make my PR useless?

Reply:
If we decide going this way yes, api changes require a bit more thought than implementation code

Reply:
Yeah that's fine, I just wanted to give you any option